From af4038a884b7d1d84dbae78f1fcee2b9a7671176 Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Mon, 26 May 2025 09:09:24 +0300
Subject: [PATCH] [SW-0] g3: move some code to sram

Move some Gaudi3 code to SRAM to free some
space in ICCM - we are near limit

Change-Id: I101ded73a92e0e474949fd7cf1e27b317a7b6d73
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 824a562..2b97239 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,6 +102,10 @@
 arch/${DEVICE}/intr/component/*.c)
 target_sources_ifdef(CONFIG_HANDLE_INTERRUPT app PRIVATE ${intr_sources})
 
+if (DEFINED CONFIG_LFS_ENABLED AND DEFINED CONFIG_LFS_OVER_SPI)
+add_compile_definitions(LFS_CONFIG=lfs_zephyr_config.h)
+add_compile_definitions(LFS_THREADSAFE=1)
+endif()
 target_sources_ifdef(CONFIG_LFS_ENABLED app PRIVATE ${ZEPHYR_MODULES_DIR}/littlefs/lfs.c
 							${ZEPHYR_MODULES_DIR}/littlefs/lfs_util.c)
 target_sources_ifdef(CONFIG_LFS_OVER_SPI app PRIVATE ${ZEPHYR_MODULES_DIR}/littlefs/zephyr/lfs_zephyrbd.c
@@ -130,4 +134,6 @@
 zephyr_code_relocate(FILES ${ZEPHYR_MODULES_DIR}/littlefs/*.c LOCATION SRAM_TEXT_RODATA)
 zephyr_code_relocate(FILES ${ZEPHYR_MODULES_DIR}/littlefs/bd/*.c LOCATION SRAM_TEXT_RODATA)
 zephyr_code_relocate(FILES ${ZEPHYR_MODULES_DIR}/littlefs/zephyr/*.c LOCATION SRAM_TEXT_RODATA)
+zephyr_code_relocate(FILES ${ZEPHYR_MODULES_DIR}/soc_init/src/${DEVICE}/skyriver_rack_eeprom.c LOCATION SRAM_TEXT_RODATA)
+zephyr_code_relocate(FILES ${ZEPHYR_MODULES_DIR}/soc_init/src/hl_serdes/${DEVICE}/serdes.c LOCATION SRAM_TEXT_RODATA)
 endif()
